### DISCUSSION TITLE
Cameras + Bots

### DISCUSSION THREAD

**Discussion by xSavgs** [Rapptz/discord.py#6358] [Ideas] [answered]
Would there ever be a way to add camera support to bots in discord.py

It might have to be a thing discord integrates but we could provide a sort of live feed to the bots camera and have it display. It would most certainly be useful.

**Comment by SebbyLaw** [comment]
I'm unsure what you mean by "camera support". Bots are unable to stream video to voice channels, and discord.py does not yet support voice/video receive.

**Comment by mesub7** [comment]
Heya.
Discord.py doesn't support and most likely won't support it in the future.
I don't think you realise how complex it is. 
See #6338.

Take care.

**Reply by xSavgs** [reply]
I understand that it's complex, and I get why discord.py wouldnt support that. I agree with you but in all essence if somehow discord came up with a way to allow it then it might not be as hard as we think.

I know their forums are basically Kickstarter ideas at this point and they barely add any of the ideas onto discord but this could be huge and it could change discord, surely we could come up with some sort of proposal to them

**Comment by Rapptz** [comment, marked as answer]
Discord does not support video streams for bots.

**Reply by xSavgs** [reply]
I understand. And I don't know how likely it would be for discord to allow bots to stream video but if somehow someone allowed this as a feature then could that be implemented?

**Comment by xSavgs** [comment]
Would the live stream feature be more realistic?

**Reply by Zomatree** [reply]
No.

**Comment by artem30801** [comment]
You can try and use something like https://pypi.org/project/pylivestream/ to livestream on youtube and use new discord feature - "voice chat activities" to play the stream in voice channel
